src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -101,20 +102,30 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
-    public function loadSettings($namespace, $ignoreUnknown = true)
+    public function load($schemaAlias, $namespace = null, $ignoreUnknown = true)
     {
-        if (isset($this->resolvedSettings[$namespace])) {
-            return $this->resolvedSettings[$namespace];
+        $schema = $this->schemaRegistry->get($schemaAlias);
+
+        $resolver = $this->defaultResolver;
+
+        if ($this->resolverRegistry->has($schemaAlias)) {
+            $resolver = $this->resolverRegistry->get($schemaAlias);
         }
 
-        if ($this->cache->contains($namespace)) {
-            $parameters = $this->cache->fetch($namespace);
-        } else {
-            $parameters = $this->getParameters($namespace);
-            $this->cache->save($namespace, $parameters);
+        // try to resolve settings for schema alias and namespace
+        $settings = $resolver->resolve($schemaAlias, $namespace);
+
+        if (!$settings) {
+            $settings = $this->settingsFactory->createNew();
+            $settings->setSchemaAlias($schemaAlias);
         }
 
-        $schema = $this->schemaRegistry->getSchema($namespace);
+        // map parameters to a plain php array
+        $parameters = $settings->getParameters()
```

Comment:
We discussed with @pamil a bit in the office and I think it would be indeed good idea to simplify this whole thing and remove Parameter model. Data is serialized so it cannot be queried against anyway. Having Settings entity with parameters array field is enough. :+1: Good idea!